Tools/ardupilotwaf/embed.py

```diff
@@ -19,48 +19,33 @@ def embed_file(out, f, idx, embedded_name, uncompressed):
     except Exception:
         raise Exception("Failed to embed %s" % f)
 
-    pad = 0
+    # ensure null termination
+    if contents[-1] != 0:
+        contents += bytes([0])
```

Comment:
this adds a 0 to the contents which means for some file types we will corrupt the file (eg. fonts).
atm we ensure there is a zero at the end for consumers that treat it as text, but the zero is after the size bytes of data

Reply:
This (trailing 0 included in the size) already happened for an uncompressed ROMFS, is that a pre-existing bug?

I had trouble understanding the requirements here anyway. How does this sound:
1. The buffer returned by `find_decompress` shall have a 0 byte within it
2. If a 0 byte has to be added to make condition 1 true, then it shall be added at the end and not be reflected in the returned `size`.

I would call it verboten anyway to read past `size` bytes in the buffer, can we fix the consumers instead?

Reply:
> is that a pre-existing bug?

yes, it is, although the boards that use uncompressed ROMFS tend to only embed the bootloader where it doesn't matter
we could look at all uses of ROMFS and see if we can avoid the zero termination requirement